### PR TITLE
Adds support for password authentication

### PR DESCRIPTION
**Pull request by janeczku**
For some people apparently password-protect their Redis instances. :wink:

### CODE REVIEW

File: pubsub/client.go [hunk 1]
```diff
@@ -122,11 +124,12 @@ const (
 )
 
 // Creates a new Radix client and subscribes to it.
-func New(address string) *Client {
+func New(address, password string) *Client {
```
Comment (ttaylorr) [inline pubsub/client.go]: Perhaps we could pass a parameter-struct that contains both (or only one) of these fields.  The runtime would transparently give the optional password field a the zero-value for its type, and you wouldn't have to specify an empty string every time you make redis clients.

I was thinking something like:
```go
package redis

type ConnectionParam struct {
        Address string
        Password string
}
```
```go
package main

import ...

func main() {
        redis := redis.New(&redis.ConnectionParam{
                Address: "localhost",
                // optional password
        })
}
```

Of course, this method would need to be changed to have the header `func (*redis.ConnectionParam) *Client`

Comment (janeczku) [inline pubsub/client.go]: Thats a much neater solution indeed. 
Why is it that Golang needs so many lines just to make a parameter optional? :grin: 
I will push this later today.

Comment (ttaylorr) [inline pubsub/client.go]: I had a similar problem the other night!  It is frustrating indeed.  Thanks in advance!

**Comment by connor4312** [comment]
Thanks for this, with Taylor's changes it should be good. I need to do some refactoring; right now redutil works but just because someone hasn't tried hard enough to break it... I'll do that after I merge your stuff in.

**Comment by janeczku** [comment]
@connor4312 @ttaylorr This should be good now. I like the true fire-and-go capability of the library. Managing long running redis connections has always been a PITA. I noticed some smaller bugs (like SubscribedEvent is not being received). But overall it *just works*.

**Comment by ttaylorr** [comment]
I'll wait for @connor4312 to merge this, but thank you for making the change!  LGTM.

**Comment by connor4312** [comment]
Looks good, thanks for contributing!

**Comment by connor4312** [comment]
@janeczku made a minor but breaking change to this in #2; had New() take a value rather than a pointer.

**Comment by janeczku** [comment]
@connor4312 thanks for the heads-up! Maybe a dumb question, but why is passing a value better?

**Comment by connor4312** [comment]
In this case, mostly because there's no need to. Pointers are good if you have a large object that gets passed around a lot -- or an object that gets tossed around and changed. But it's more idiomatic to use values when that's not the case (see the [code review comments](https://github.com/golang/go/wiki/CodeReviewComments#pass-values)). It's unfortunate that their wiki doesn't mention the _why_, and I'm hardly a master Gopher, but here are some reasons I have found:

 * When passing a value, you have a guarantee in that the value will never change. In this case, that's useful. I don't have to deal with people changing their connection params in the middle of the connection. On a similar note:
 * A function that takes a pointer receiver can change the underlying value. For example, if I'm passing in my application's main config struct, I probably want to be sure that no one changes it, because doing so could cause really weird behaviour. It's less cognitive load to be aware of.
 * Values are likely to get allocated on the stack, while pointers are usually allocated on the heap, so there's a minor performance difference. In this case it'll be allocated on the heap anyway, but as a general rule...
 * It's cleaner.

**Comment by janeczku** [comment]
This gave me a much better understanding about the implications of using pointers vs. values in function receivers. Made it clear to me that any considerations on performance or memory allocation should always come last in the decision chain. Much appreciated!

**Pull request merged**